deps/rabbit/src/rabbit_disk_usage.erl

```diff
@@ -0,0 +1,76 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%% Copyright (c) 2007-2025 Broadcom. All Rights Reserved. The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries. All rights reserved.
+%%
+
+-module(rabbit_disk_usage).
+
+%% Functions for calculating disk usage of a given directory.
+
+-include_lib("kernel/include/file.hrl").
+
+-export([scan/1]).
+
+%% @doc Calculates the disk usage in bytes of the given directory.
+%%
+%% On especially large directories this can be an expensive operation since
+%% each sub-directory is scanned recursively and each file's metadata must be
+%% read.
```

Comment:
Might be better for queue types to increment/decrement an internal counter (that may be stored in persistent_term).